src/core/site.jl

```diff
@@ -40,11 +40,25 @@ function Site(;
     roof_squarefeet::Union{Real, Nothing} = nothing,
     min_resil_time_steps::Int=0,
     mg_tech_sizes_equal_grid_sizes::Bool = true,
+    CO2_emissions_reduction_min_pct::Union{Float64, Nothing} = nothing,
+    CO2_emissions_reduction_max_pct::Union{Float64, Nothing} = nothing,
+    bau_emissions_lb_CO2_per_year::Union{Float64, Nothing} = nothing,
```

Comment:
Actually I wasn't intending this to be a possible user input. These fields should be in the Site struct but not in its constructor. I'm a bit wary of users being able to provide BAU emissions because if they use a different emissions dataset to calculate them then it wouldn't be an apples to apples comparison. But at the same time, it could definitely be useful if used correctly and would allow for emissions reduction constraints without running the BAU model. Maybe a compromise is allowing them as user inputs in the Julia package but not int he API?

Reply:
it seems like BAU emissions should not allow the Nothing type, nor be part of Site. They are only needed in REoptInputs I think?  What are the cases that we need to calculate BAU emissions? Every run? Perhaps we can reduce overhead by adding a Emissions input struct that when present in the user inputs triggers all of the emissions capabilities?

Reply:
Here is a snapshot of where major emissions inputs reside in REopt as-is. Adding a new emissions struct where we always run some functionality and conditionally (i.e. take different actions if off-grid is true etc.) run some functionality makes sense as well.
![image](https://user-images.githubusercontent.com/83797218/176244880-53a35f35-fa7b-43f0-bdea-0d0b3a4d7adb.png)

Reply:
BAU emissions need to be nothing if there is no BAU scenario.

Reply:
I did consider the idea of creating a separate emissions struct, but opted to keep emissions inputs in their respective structs rather than overhauling the approach that @adfarth and Kathleen took in the API.

Reply:
I am tracking. there is a lot of results processing being done in `results/site.jl` for emissions. However the fields related to emissions in the Site struct appear to be set in the Site struct for BAU and optimal scenarios and aren't used downstream in emissions reduction or pct RE electricity calculations. Instead we are currently creating new keys in `add_site_results` function. So maybe we have some room to take out any fields in the Site structs which arent used downstream in calculations or model setting?

Reply:
True, it's totally unnecessary for us to be calculating BAU emissions before optimization for any pollutants other than CO2 because that's the only one with a reduction % constraint. I deleted all of that in 4fe60b2